### PR TITLE
Fix getcompletion() "cmdline" not working for command after :autocmd

### PR DESCRIPTION
**Pull request by zeertzjq**
Problem:    getcompletion() "cmdline" fails for command after :autocmd.
Solution:   Use set_cmd_context() instead of set_one_cmd_context().

### CODE REVIEW

File: src/usercmd.c [hunk 1]
```diff
@@ -329,11 +329,7 @@ set_context_in_user_cmdarg(
 	return set_context_in_menu_cmd(xp, cmd, arg, forceit);
 #endif
     if (context == EXPAND_COMMANDS)
-    {
-	if (xp->xp_context == EXPAND_NOTHING)
-	    xp->xp_context = context;
```
Comment (yegappan) [inline src/usercmd.c:334]: Is this change needed?  Will this break something else?

Comment (zeertzjq) [reply src/usercmd.c:334]: It was just added two days ago to fix a problem with `getcompletion()`.

Comment (zeertzjq) [reply src/usercmd.c:334]: After this PR, `set_one_cmd_context()` is only called by `set_cmd_context()`.

When this branch is reached, `set_context_in_user_cmdarg()` returns non-`NULL` to `set_context_by_cmdname()` -> `set_one_cmd_context()` -> `set_cmd_context()`, and `set_cmd_context()` calls `set_one_cmd_context()` again, which sets `xp->xp_context = EXPAND_COMMAND` immediately. Therefore this assignment is no longer needed.

**Review by yegappan — APPROVED** [review]

File: src/usercmd.c (hunk 1 above)
Comment (chrisbra) [reply src/usercmd.c:334]: okay thanks
